src/Microsoft.AspNetCore.ResponseCaching/ResponseCachingContext.cs

```diff
@@ -143,49 +145,61 @@ internal string CreateCacheKey()
 
         internal string CreateCacheKey(CachedVaryBy varyBy)
         {
-            var request = HttpContext.Request;
-            var builder = new StringBuilder()
-                .Append(request.Method.ToUpperInvariant())
-                .Append(";")
-                .Append(request.Path.Value.ToUpperInvariant());
+            var request = _httpContext.Request;
+            var builder = _builderPool?.Get() ?? new StringBuilder();
```

Comment:
One less visible caveat is that the default StringBuilder pool will has a max capacity of 4*1024 characters. If the generated key is larger than that value, no pooling will occur and we are back to allocating one per request. However, The user can replace the default policy to increase the max capacity. Also, 4*1024 characters is quite a large value so I'm not sure how likely this limit is hit.